src/reconcile/rec_visibility.c

```diff
@@ -414,12 +414,51 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
 
             /* Find the update this tombstone applies to. */
             if (!__wt_txn_upd_visible_all(session, upd)) {
-                while (upd->next != NULL && upd->next->txnid == WT_TXN_ABORTED)
-                    upd = upd->next;
+                /*
+                 * Loop until a valid update from a different transaction is found in the update
+                 * list.
+                 */
+                while (upd->next != NULL) {
+                    if (upd->next->txnid == WT_TXN_ABORTED)
+                        upd = upd->next;
+                    else if (tombstone->txnid == upd->next->txnid &&
+                      tombstone->start_ts == upd->next->start_ts &&
```

Comment:
@kommiharibabu 
I think it's the same here right?
You can probably just assert that the timestamps match up.

Reply:
Yes, you are right. Fixed here also and added asserts to check for the timestamp values.

Reply:
@tetsuo-cpp It is possible that a single transaction with multiple timestamped updates. Refer test_timestamp06.py, due to which I removed the assert I added in the above function. But with prepare transaction it is not possible to have a single update with multiple prepare timestamps. So I kept the assert in the prepared flow of bt_page.c

Reply:
@kommiharibabu Ah understood. You're right.